core/src/main/resources/error/error-classes.json

```diff
@@ -1326,6 +1326,11 @@
       "grouping()/grouping_id() can only be used with GroupingSets/Cube/Rollup"
     ]
   },
+  "UNSUPPORTED_LITERAL_FOR_SOURCE_TYPE" : {
+    "message" : [
+      "Literals of the type <field> are currently not supported for the <sourceDataType> type."
```

Comment:
Looking at where the error comes from:
1. https://github.com/apache/spark/blob/51e4c2cc55aa01f07b28b1cd807b553f8729075d/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/intervalExpressions.scala#L159
2. https://github.com/apache/spark/blob/b14da8b1b65d9f00f49fab87f738715089bc43e8/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/datetimeExpressions.scala#L2795

, the error message and error class name confuse slightly. The error is mostly about an extract field. How about error class name:
`INVALID_EXTRACT_FIELD`
Also cc @srielau

Reply:
OK, Let me to do it.